custom_components/tuya_local/devices/aygrochy_inline_wifi_air_vent.yaml

```diff
@@ -0,0 +1,301 @@
+name: Aygrochy fan
+primary_entity:
+  entity: climate
+  dps:
+    - id: 2
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: "ON"
+          value: "ON"
+        - dps_val: "OFF"
+          value: "OFF"
+        - dps_val: "TIMER"
+          value: "TIMER"
+        - dps_val: "AUTO"
+          value: "AUTO"
+        - dps_val: "ALARM"
+          value: "ALARM"
+    - id: 2
+      name: hvac_mode
+      type: string
+      mapping:
+        - dps_val: "ON"
+          value: "fan_only"
+        - dps_val: "OFF"
+          value: "off"
+        - dps_val: "TIMER"
+          value: "auto"
+          hidden: true
+        - dps_val: "AUTO"
+          value: "auto"
+        - dps_val: "ALARM"
+          value: "fan_only"
+          hidden: true
+    - id: 8
+      name: current_humidity
+      type: integer
+      unit: "%"
+    - id: 9
+      name: current_temperature
+      type: integer
+      unit: "F"
+    - id: 106
+      name: target_temp_high
+      type: integer
+      range:
+        min: 32
+        max: 194
+      unit: "F"
+    - id: 108
+      name: target_temp_low
+      type: integer
+      range:
+        min: 32
+        max: 194
+      unit: "F"
+
+secondary_entities:
+  - entity: fan
+    dps:
+      - id: 1
+        name: switch
+        type: boolean
+      - id: 102
+        name: speed
+        type: integer
+        range:
+          min: 0
+          max: 10
+  - entity: light
+    translation_key: backlight
+    dps:
+      - id: 101
+        name: brightness
+        type: integer
+        range:
+          min: 0
+          max: 4
+  - entity: sensor
+    name: Current speed
+    dps:
+      - id: 103
+        name: sensor
+        type: integer
+  - entity: number
+    translation_key: timer
+    dps:
+      - id: 104
+        name: value
+        type: integer
+        unit: min
+        range:
+          min: 0
+          max: 36000
+        mapping:
+          - scale: 60
+            step: 60
+  - entity: sensor
+    name: Alarm triggered
+    dps:
+      - id: 105
+        name: sensor
+        type: string
+  # id 106 see primairy entity
+  - entity: number
+    name: Auto-High Humid
+    dps:
+      - id: 107
+        name: value
+        class: humidity
```

Comment:
This class needs to go up at the entity level

Reply:
fixed